Modules/_heapqmodule.c

```diff
@@ -172,8 +173,10 @@ heappop_internal(PyObject *heap, int siftup_func(PyListObject *, Py_ssize_t))
     if (!n)
         return lastelt;
     returnitem = PyList_GET_ITEM(heap, 0);
-    PyList_SET_ITEM(heap, 0, lastelt);
-    if (siftup_func((PyListObject *)heap, 0)) {
+    // We're in the critical section now
```

Comment:
This comment is incorrect in the sense that the critical section is held from the very beginning so it not "now"